File: Ocampo-Final-CS103/FinalsOcampoCedrik.py
class Review:
    def __init__(self, name, feedback, rating):
        self.name = name
        self.feedback = feedback
        self.rating = rating

def add_review(reviews):
    print("==================================")
    print("---------Leave a Review-----------\n")
    name = input("Enter your name: ")
    feedback = input("Enter your feedback: ")

    while True:
        try:
            rating = int(input("Enter your rating (1-5): "))
            if 1 <= rating <= 5:
                break
            else:
                print("Invalid rating. Please enter a number between 1 and 5.")
        except ValueError:
            print("Invalid input. Please enter a number between 1 and 5.")

    reviews.append(Review(name, feedback, rating))

def display_reviews(reviews):
    if not reviews:
        print("No reviews yet.")
        return

    total_rating = 0
    print("==================================")
    print("-------------Reviews--------------\n")
    for review in reviews:
        print(f"Name: {review.name}")
        print(f"Feedback: {review.feedback}")
        print(f"Rating: {review.rating}/5")
        print("----------------------------------")
        total_rating += review.rating

    average_rating = total_rating / len(reviews)
    print(f"Average Rating: {average_rating:.1f}/5")

def main():
    reviews = []
    while True:
        print("==================================")
        print("--------Restaurant Review---------\n")
        print("1. Add Review")
        print("2. Display Reviews")
        print("3. Exit\n")
        print("--------(Input no.: 1 - 3)--------")
        try:
            choice = int(input("Enter your choice: "))
            if choice == 1:
                add_review(reviews)
            elif choice == 2:
                display_reviews(reviews)
            elif choice == 3:
                print("Exiting...")
                break
            else:
                print("Invalid choice. Please enter a number between 1 and 3.")
        except ValueError:
            print("Invalid choice. Please enter a number between 1 and 3.")

if __name__ == "__main__":
    main()

File: Ocampo-Final-CS103/FinalsOcampoCedrik.cpp
#include <iostream>
#include <vector>
#include <string>
#include <limits>

using namespace std;

struct Review {
    string name;
    string feedback;
    int rating;
};

void addReview(vector<Review>& reviews) {
    Review newReview;
    cout << "==================================\n";
    cout << "---------Leave a Review-----------\n\n";
    cout << "Enter your name: ";
    cin.ignore();
    getline(cin, newReview.name);
    cout << "Enter your feedback: ";
    getline(cin, newReview.feedback);

    while (true) {
        cout << "Enter your rating (1-5): ";
        cin >> newReview.rating;
        if (cin.fail() || newReview.rating < 1 || newReview.rating > 5) {
            cin.clear(); // clear the error flag
            cin.ignore(numeric_limits<streamsize>::max(), '\n'); // discard invalid input
            cout << "Invalid rating. Please enter a number between 1 and 5." << endl;
        }
        else {
            break;
        }
    }

    reviews.push_back(newReview);
}

void displayReviews(const vector<Review>& reviews) {
    if (reviews.empty()) {
        cout << "No reviews yet." << endl;
        return;
    }

    double totalRating = 0;
    cout << "==================================\n";
    cout << "-------------Reviews--------------\n\n";
    for (const auto& review : reviews) {
        cout << "Name: " << review.name << endl;
        cout << "Feedback: " << review.feedback << endl;
        cout << "Rating: " << review.rating << "/5" << endl;
        cout << "----------------------------------" << endl;
        totalRating += review.rating;
    }

    double averageRating = totalRating / reviews.size();
    cout << "Average Rating: " << averageRating << "/5" << endl;
}

int main() {
    vector<Review> reviews;
    int choice;

    do {
        cout << "==================================\n";
        cout << "--------Restaurant Review---------\n\n";
        cout << "1. Add Review" << endl;
        cout << "2. Display Reviews" << endl;
        cout << "3. Exit\n\n";
        cout << "--------(Input no.: 1 - 3)--------\n";
        cout << "Enter your choice: ";
        cin >> choice;

        if (cin.fail() || choice < 1 || choice > 3) {
            cin.clear(); // clear the error flag
            cin.ignore(numeric_limits<streamsize>::max(), '\n'); // discard invalid input
            cout << "Invalid choice. Please enter a number between 1 and 3." << endl;
            continue;
        }

        switch (choice) {
        case 1:
            addReview(reviews);
            break;
        case 2:
            displayReviews(reviews);
            break;
        case 3:
            cout << "Exiting..." << endl;
            break;
        }
    } while (choice != 3);

    return 0;
}
